src/dynamic_bridge.cpp

```diff
@@ -192,13 +192,17 @@ void update_bridge(
         bridge.ros1_type_name, topic_name, 10,
         bridge.ros2_type_name, topic_name, ros2_publisher_qos);
     } catch (std::runtime_error & e) {
-      fprintf(
-        stderr,
-        "failed to create 1to2 bridge for topic '%s' "
-        "with ROS 1 type '%s' and ROS 2 type '%s': %s\n",
-        topic_name.c_str(), bridge.ros1_type_name.c_str(), bridge.ros2_type_name.c_str(), e.what());
-      if (std::string(e.what()).find("No template specialization") != std::string::npos) {
-        fprintf(stderr, "check the list of supported pairs with the `--print-pairs` option\n");
+      static std::set<std::string> logged_topic_errors;
```

Comment:
having this static global variable is safe since both ros1 and ros2 uses single thread to poll the process.